src/main/java/com/snowflake/kafka/connector/SnowflakeSinkTask.java

```diff
@@ -388,29 +388,31 @@ private static void waitFor(Supplier<Boolean> func)
     throw new TimeoutException();
   }
 
-  private static long getExecutionTimeSec(long startTime, long endTime) {
-    return (endTime - startTime) / 1000;
+  private static long getDurationFromStartMs(long startTime) {
+    final long currTime = System.currentTimeMillis();
+    return currTime - startTime;
   }
 
   void logWarningForPutAndPrecommit(long startTime, int size, String apiName) {
-    long executionTime = getExecutionTimeSec(startTime, System.currentTimeMillis());
-    if (executionTime > 300) {
+    long executionTimeMs = getDurationFromStartMs(startTime);
+    if (executionTimeMs > 300000) {
       // This won't be frequently printed. It is vary rare to have execution greater than 300
       // seconds.
       // But having this warning helps customer to debug their Kafka Connect config.
       this.DYNAMIC_LOGGER.warn(
-          "{} {}. Time: {} seconds > 300 seconds. If there is CommitFailedException in the log or"
-              + " there is duplicated records, refer to this link for solution: "
+          "{} {}. Time: {} ms = {} seconds > 300 seconds. If there is CommitFailedException in the"
+              + " log or there is duplicated records, refer to this link for solution: "
               + "https://docs.snowflake.com/en/user-guide/kafka-connector-ts.html#resolving-specific-issues",
           apiName,
           size,
-          executionTime);
+          executionTimeMs,
+          executionTimeMs / 1000);
     } else {
-      this.DYNAMIC_LOGGER.debug(
-          "successfully called {} with {} records, execution time: {} seconds",
+      this.DYNAMIC_LOGGER.info(
+          "successfully called {} with {} records, execution time: {} ms",
           apiName,
           size,
```

Comment:
can you fix the records thing as well? just may be check apiName and if it is precommit, print partitions. (It's hacky but works for now)

Reply:
Yup, it now prints the total number of offsets and the number of commitable offsets. updated the description with examples of the logs